tests/cxx/typedef_chain_in_template-d4.h

```diff
@@ -0,0 +1,25 @@
+//===--- typedef_chain_in_template-d4.h - test input file for iwyu --------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// Container that tests typedef chains longer than in libc++.
+
+template<typename T>
+class ContainerLongTypedefChain {
+ public:
+  typedef T value_type;
+  typedef value_type& reference;
+  typedef reference& reference_reference;
```

Comment:
I think `reference_reference` is not a rvalue. Don't remember exactly but AST doesn't look like rvalue. And also when I run `clang -fsyntax-only -std=c++03 -I . tests/cxx/typedef_chain_in_template.cc` or `clang -fsyntax-only -std=c++98 -I . tests/cxx/typedef_chain_in_template.cc` - clang doesn't complain.

I wasn't trying to test rvalues, these are my frantic attempts to trigger execution of
```c++
if (underlying_typedef->getDecl()->getLexicalDeclContext() == parent) {
  return GetFirstNonTemplateTypedef(underlying_typedef, parent);
}
```

I decided to leave this test case for more complete test suite and to test that cases like this don't confuse IWYU.